src/pybind/mgr/ansible/ansible_runner_svc.py

```diff
@@ -0,0 +1,272 @@
+
+        if self.play_uuid == '-': # Initialized
+            status_value = ExecutionStatusCode.NOT_LAUNCHED
+        elif self.play_uuid == '': # Error launching playbook
+            status_value = ExecutionStatusCode.ERROR
+        else:
+            endpoint = "%s/%s" % (PLAYBOOK_EXEC_URL, self.play_uuid)
+            response = self.rest_client.http_get(endpoint)
+
+            if response:
+                the_status = json.loads(response.text)["msg"]
+                if the_status == 'successful':
+                    status_value = ExecutionStatusCode.SUCCESS
+                elif the_status == 'failed':
+                    status_value = ExecutionStatusCode.ERROR
+                else:
+                    status_value = ExecutionStatusCode.ON_GOING
+            else:
+                status_value = ExecutionStatusCode.ERROR
+
+        self.log.info("Requested playbook execution status is: %s", status_value)
+        return status_value
+
+    def get_result(self, event_filter=""):
+        """Get the data of the events filtered by a task pattern and
+        a event filter
+
+        @returns: the events that matches with the patterns provided
+        """
+
+        if not self.result_task_pattern or not self.play_uuid:
+            result_events = {}
+
+        response = self.rest_client.http_get(PLAYBOOK_EVENTS % self.play_uuid)
+
+        if not response:
+            result_events = {}
+        else:
+            events = json.loads(response.text)["data"]["events"]
+            result_events = {event:data for event,data in events.items()
+                            if "task" in data and
+                            re.match(self.result_task_pattern, data["task"])}
+            if event_filter:
+                result_events = {event:data for event,data in result_events.items()
+                                if re.match(event_filter, data['event'])}
+
+        self.log.info("Requested playbook result is: %s", json.dumps(result_events))
+        return result_events
+
+class Client(object):
+    """An utility object that allows to connect with the Ansible runner service
+    and execute easily playbooks
+    """
+
+    def __init__(self, server_url, user, password, certificate, logger):
+        """Provide an https client to make easy interact with the Ansible
+        Runner Service"
+
+        @param servers_url: The base URL >server>:<port> of the Ansible Runner Service
+        @param user: User name of the authorized user
+        @param password: Password of the authotized user
+        @param certificate: Certificate file of the Ansible Runner Service
+        @param logger: Log file
+        """
+        self.server_url = server_url
+        self.user = user
+        self.password = password
+        self.log = logger
+        self.auth = (self.user, self.password)
+
+        # Used to verify or not https server identity
+        if not certificate:
+            self.certificate = False
```

Comment:
Can we set the default to `True`? Disabling HTTPS validation is questionable.

Reply:
There is a setting ( "certificate") config to specify the path to the CA Bundle to use for verification. 
If this is not provided then is assumed that we cannot verify the Ansible Server Identity, (this would be like a "dev" mode.)
- BTW, probably the name of the config setting is not the best one.... -

In any case, in the Ansible Runner Service there is a change to be implement for using client TLS certificates. 
https://github.com/pcuzner/ansible-runner-service/issues/74
And this probably will imply changes in the login method of the orchestrator.

Reply:
> There is a setting ( "certificate") config to specify the path to the CA Bundle to use for verification. If this is not provided then is assumed that we cannot verify the Ansible Server Identity, (this would be like a "dev" mode.)

Instead of assuming per default that we cannot verify the Identity, would it be possible to let the user explicitly disable verification?

Reply:
Ok ... safe by default... i got it. Sure!!. I will change it asap. thx!